go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/elastic/elastic-agent
 
-go 1.21
+go 1.21.0
+
+toolchain go1.21.8
```

Comment:
go version is specified as `1.21.9` maybe we should match and update bump go version to update go.mod as well

Reply:
Fixed (manually) in 176df272f5a8f0393c2eabc202e51a4491d13458.  

Put up https://github.com/elastic/elastic-agent/pull/4636 to automatically keep the `toolchain` version here in sync with the contents of `.go-version`.